agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -201,6 +201,7 @@ def start(info)
         end
       end
 
+      attach_router unless interface_ip('weave')
```

Comment:
If weave interface exist without address and agent heals it (with a rescue from `Errno::EADDRNOTAVAIL`) for some reason the "healed" weave interface seems to be slightly different compared when it's completely re-created:
```
core@quiet-frost-93 ~ $ ip addr show dev weave
8: weave: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1410 qdisc noqueue state UP group default qlen 1000
    link/ether be:f4:ed:2d:50:04 brd ff:ff:ff:ff:ff:ff
    inet 10.81.0.1/19 scope global weave
       valid_lft forever preferred_lft forever
```
vs. normally created interface:
```
core@quiet-frost-93 ~ $ ip addr show dev weave
8: weave: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1410 qdisc noqueue state UP group default qlen 1000
    link/ether be:f4:ed:2d:50:04 brd ff:ff:ff:ff:ff:ff
    inet 10.81.0.1/19 scope global weave
       valid_lft forever preferred_lft forever
    inet6 fe80::bcf4:edff:fe2d:5004/64 scope link 
       valid_lft forever preferred_lft forever
```

So it misses IPv6 config. As weave (and Kontena) doesn't support IPv6 this should not be an issue.